core/src/main/java/io/fabric8/maven/core/util/MavenUtil.java

```diff
@@ -246,4 +251,56 @@ public static void createArchive(File sourceDir, File destinationFile, JarArchiv
             throw new MojoExecutionException("Failed to create archive " + destinationFile + ": " + e, e);
         }
     }
+
+    /**
+     * Returns the version from the list of pre-configured versions of common groupId/artifact pairs
+     */
+    public static String getVersion(String groupId, String artifactId) {
+        String key = "" + groupId + "/" + artifactId;
+        Map<String, String> map = getGroupArtifactVersionMap();
+        String version = map.get(key);
+        if (version == null) {
+            LOG.warn("Could not find the version for groupId: " + groupId + " artifactId: " + artifactId + " in: " + map);
+        }
+        return version;
+    }
+
+    /**
+     * Returns the version from the list of pre-configured versions of common groupId/artifact pairs
+     */
+    public static String getVersion(String groupId, String artifactId, String defaultVersion) {
+        String answer = getVersion(groupId, artifactId);
+        if (Strings.isNullOrBlank(answer)) {
+            answer = defaultVersion;
+        }
+        return answer;
+    }
+
+    protected static Map<String, String> getGroupArtifactVersionMap() {
+        if (groupArtifactVersionMap == null) {
+            groupArtifactVersionMap = new HashMap<>();
+
+            InputStream in = MavenUtil.class.getResourceAsStream("versions.properties");
```

Comment:
I wonder whether it wouldn't be better to pick up the version we need from `META-INF/maven/${groupId}/${artifactId}/pom.properties` ? This is alway generated by Maven so we don't need any extra custom version properties. See also http://stackoverflow.com/questions/3697449/retrieve-version-from-maven-pom-xml-in-code